### PR TITLE
Add maturin

### PR DESCRIPTION
**Pull request by apcamargo**
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by apcamargo** [comment]
@conda-forge/help-python-c @conda-forge/help-python 

The recipe builds fine in Linux and in Windows, but I get the following errors in OSX:

```
Packaging maturin
INFO conda_build.build:build(1571): Packaging maturin
Packaging maturin-0.7.7-py36h0b31af3_0
INFO conda_build.build:bundle_conda(891): Packaging maturin-0.7.7-py36h0b31af3_0
compiling .pyc files...
number of files: 8
Fixing linking of /System/Library/Frameworks/Security.framework/Versions/A/Security in /usr/local/miniconda/conda-bld/maturin_1582233876270/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/maturin
New link location is None
Fixing linking of /usr/lib/libSystem.B.dylib in /usr/local/miniconda/conda-bld/maturin_1582233876270/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/maturin
New link location is None
Fixing linking of /usr/lib/libresolv.9.dylib in /usr/local/miniconda/conda-bld/maturin_1582233876270/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_place/bin/maturin
New link location is None

  ERROR (maturin,bin/maturin): /usr/lib/libresolv.9.dylib not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?
   INFO (maturin,bin/maturin): Needed DSO /System/Library/Frameworks/Security.framework/Versions/A/Security found in the whitelist
   INFO (maturin,bin/maturin): Needed DSO /usr/lib/libSystem.B.dylib found in the whitelist
Traceback (most recent call last):
  File ".ci_support/build_all.py", line 135, in <module>
    build_all(args.recipes_dir, args.arch)
  File ".ci_support/build_all.py", line 68, in build_all
    build_folders(recipes_dir, new_comp_folders, arch, channel_urls)
  File ".ci_support/build_all.py", line 125, in build_folders
    conda_build.api.build([recipe], config=get_config(arch, channel_urls))
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/api.py", line 209, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/build.py", line 2344, in build_tree
    notest=notest,
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/build.py", line 1643, in build
    newly_built_packages = bundlers[pkg_type](output_d, m, env, stats)
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/build.py", line 1002, in bundle_conda
    files = post_process_files(metadata, initial_files)
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/build.py", line 867, in post_process_files
    post_build(m, new_files, build_python=python)
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/post.py", line 1150, in post_build
    check_overlinking(m, files)
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/post.py", line 1087, in check_overlinking
    m.config.enable_static)
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/post.py", line 1051, in check_overlinking_impl
    raise OverLinkingError(overlinking_errors)
conda_build.exceptions.OverLinkingError: overlinking check failed 
['  ERROR (maturin,bin/maturin): /usr/lib/libresolv.9.dylib not found in sysroot, is this binary repackaging? .. do you need to use install_name_tool/patchelf?']
```

**Comment by scopatz** [comment]
Maybe this is from the rust host dep?

**Comment by apcamargo** [comment]
@scopatz I don't think I follow. Why would this be a problem? I tried to do the same as other rust recipes.

**Comment by scopatz** [comment]
Or maybe you have an undeclared dependency that lives on the system?

**Comment by apcamargo** [comment]
@conda-forge/staged-recipes Anyone with OSX expertise that knows what may be going on here?

File: recipes/maturin/build.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+set -ex
+
+if [ `uname` == Darwin ]; then
+  export MACOSX_DEPLOYMENT_TARGET=10.13
```
Comment (isuruf) [inline recipes/maturin/build.sh]: Is this needed?

**Comment by isuruf** [comment]
You need https://github.com/conda-forge/cyrus-sasl-feedstock/blob/9a84fd3bdf037c891971f374813d1829eab86cbd/recipe/meta.yaml#L27-L28

File: recipes/maturin/build.sh (hunk 1 above)
Comment (apcamargo) [reply recipes/maturin/build.sh]: It's not. I'll remove it in the next commit

**Comment by apcamargo** [comment]
It worked! Thank you @isuruf!

File: recipes/maturin/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 2533c648aa8fa19f7a9d4f62282b4adce2f9c07c49f512005f92911be767555e
+
+build:
+  number: 0
+  skip: True  # [py27]
+  missing_dso_whitelist:          # [osx]
+    - /usr/lib/libresolv.9.dylib  # [osx]
+    - /usr/lib/libpam.2.dylib     # [osx]
```
Comment (isuruf) [inline recipes/maturin/meta.yaml]: Do you need this line?

File: recipes/maturin/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+test:
+  commands:
+    - maturin --help
+
+about:
+  home: https://github.com/PyO3/maturin
+  license: MIT
+  license_family: MIT
+  license_file: license-mit
+  summary: "Build and publish crates with pyo3, rust-cpython and cffi bindings as well as rust binaries as python packages"
```
Comment (jakirkham) [inline recipes/maturin/meta.yaml]: Would suggest wrapping this text. Possibly like so?

```suggestion
  summary: >-
    Build and publish crates with pyo3, rust-cpython and cffi bindings
    as well as rust binaries as python packages
```

File: recipes/maturin/meta.yaml (hunk 2 above)
Comment (apcamargo) [reply recipes/maturin/meta.yaml]: I copied it by mistake. Just removed it.

Comment (apcamargo) [reply recipes/maturin/meta.yaml]: Thank you!

File: recipes/maturin/meta.yaml (hunk 3 above)
Comment (apcamargo) [reply recipes/maturin/meta.yaml]: Thank you!

File: recipes/maturin/license-mit [hunk 4]
```diff
@@ -0,0 +1,25 @@
+Copyright (c) 2018 konstin
```
Comment (jakirkham) [inline recipes/maturin/license-mit]: Can drop this file. It appears to already be in the source we are downloading.

Comment (apcamargo) [reply recipes/maturin/license-mit]: Done

**Comment by apcamargo** [comment]
Thank you, @jakirkham 

@conda-forge/staged-recipes I think this is ready for a merge.

**Comment by apcamargo** [comment]
Ready for review and merge @conda-forge/help-python

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**